### ISSUE TITLE
Binary XCFrameworks do not contain debug symbols

### ISSUE THREAD

**Issue by ffittschen** [aws-amplify/aws-sdk-ios#3562]
**Describe the bug**
The binary xcframeworks distributed with Swift Package Manager via https://github.com/aws-amplify/aws-sdk-ios-spm do not contain any debug symbols.

**To Reproduce**
1. Integrate any target of the AWS SDK into an app that uses some crash reporter, such as Visual Studio AppCenter
2. Trigger a crash within the AWS SDK
3. Look at the crash logs in the crash reporter
4. See the hidden stack traces

**Observed Behavior**
The stack traces are not desymbolicated, which makes them useless.

**Expected Behavior**
I'd expect that the binary xcframeworks contain dSYMs (and bitcode) since this is supported since Xcode 12.

**Stack Trace**
```
Thread 10 Crashed:
0   CoreFoundation                       0x000000019689cb38 0x196868000 + 215864
1   CoreFoundation                       0x0000000196921df0 0x196868000 + 761328
2   AWSIoT                               0x0000000107b13118 _hidden#41_ (__hidden#478_:650)
3   AWSIoT                               0x0000000107b16d34 _hidden#79_ (__hidden#478_:1232)
4   AWSIoT                               0x0000000107b8cea0 _hidden#15137_ (__hidden#15366_:1610)
5   libdispatch.dylib                    0x000000019658024c 0x19657e000 + 8780
6   libdispatch.dylib                    0x0000000196581db0 0x19657e000 + 15792
7   libdispatch.dylib                    0x0000000196592a68 0x19657e000 + 84584
8   libdispatch.dylib                    0x0000000196593120 0x19657e000 + 86304
9   libsystem_pthread.dylib              0x00000001e24397d8 0x1e2436000 + 14296
10  libsystem_pthread.dylib              0x00000001e244076c 0x1e2436000 + 42860
```

**Screenshots**

![Screen Shot 2021-04-28 at 10 44 22@2x](https://user-images.githubusercontent.com/7734806/116376688-c865f400-a810-11eb-8c00-a22229aefaae.png)

**Environment(please complete the following information):**
 - SDK Version: 2.23.3
 - Dependency Manager: Swift Package Manager
 - Swift Version: 5.4
 - Xcode Version: 12.5

**Additional context**
We integrated `AWSIoT` and `AWSCore` according to the [README-spm-support.md](https://github.com/aws-amplify/aws-sdk-ios/blob/main/README-spm-support.md) file.

**Comment by brennanMKE** [comment]
Updated CircleCi script to include debug files and confirmed with FileMerge between the output before and after the update.

<img width="1129" alt="XCF-Debug" src="https://user-images.githubusercontent.com/277419/120872613-0ebb1b00-c554-11eb-8b6e-db090de69327.png">

**Comment by diegocstn** [comment]
dSYMs are included in binary XCFrameworks since version v2.24.3 of the SDKs.
